### PR TITLE
Fix false positives in Usage sanity checks of templates and network offerings 

### PR DESCRIPTION
**Pull request by winterhazel**
### Description

This PR addresses two false positives in the Usage sanity checker:

(1)

A sanity check verifies if there are usage records of a template/ISO created after the template/ISO was removed.

In an environment with two zones, if a template gets removed from a zone and added back to it, `cloud.template_zone_ref` will have two entries related to the template; one marked as `removed`. In this situation, the operator will receive a false positive stating that there are usage records for the template/ISO after it was removed, even though the template/ISO is available in that zone.

This check was changed to verify if the template/ISO is currently available in that zone.

(2)

A sanity check verifies if there are usage records with a raw usage greater than the aggregation range.

If a VM has more than one NIC using the same network offering, CloudStack will group the usage of both NICs into a single network offering usage record; for example, two NICs used for 1 hour will result in an entry with a raw usage of 2 hours. In this situation, the operator will receive a false positive stating that there are usage records with a raw usage greater than the aggregation range.

This check was changed to consider the average usage of the NICs, instead of the raw usage.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?

I changed the initial delay in `UsageManagerImpl.java#L315` to 0 in order to run the sanity checks immediately when Usage starts. Then, whenever I wanted to run the sanity checks, I would set the last checked id in `/usr/local/libexec/sanity-check-last-id` to 1 and restart Usage.

To simulate the scenario in (1), I created the zone `zone2`, copied a template that was being used in `zone1` to `zone2` and deleted the template from `zone1`. Then, I copied it from `zone2` to `zone1`.

Before applying the changes:
- I ran the sanity checks and verified that CloudStack reported template/ISO usage records created after it was removed;

After applying the changes
- I ran the sanity checks and verified that no errors were found;
- I deleted the template from `zone1`, ran the sanity checks again and verified that CloudStack reported template/ISO usage records created after it was removed.

To simulate the scenario in (2), I set `usage.stats.job.aggregation.range` to 60 (hourly), created a VM with two NICs and waited until there were network offering usage records corresponding to the NICs. I verified that `raw_usage` was indeed 2.

Before applying the changes:
- I ran the sanity checks and verified that CloudStack reported usage records with `raw_usage` greater than 1.

After applying the changes:
- I ran the sanity checks and verified that CloudStack did not report usage records with `raw_usage` greater than 1;
- I manually changed the `raw_usage` of a usage record from 2 to 3, ran the sanity checks again and verified that CloudStack reported network offering usage records with `raw_usage` greater than 1.

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
good fixes @winterhazel 
clgtm

**Comment by DaanHoogland** [comment]
@winterhazel , you can add yourself to the contributers list in .asf.yaml
Just create a PR for us to merge. It will give you some extra possibilities, like automatically run the github actions and being able to label PRs.

**Review by rohityadavcloud — APPROVED** [review]
LGTM but needs additional review/testing

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma8 kvm-alma8

**Comment by winterhazel** [comment]
@DaanHoogland @harikrishna-patnala @shwstppr could you guys please validate the changes in this PR for merging? Thank you.

**Review by shwstppr — APPROVED** [review]
Code LGTM. Needs testing

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by winterhazel** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

File: usage/src/main/java/com/cloud/usage/UsageSanityChecker.java [hunk 1]
```diff
@@ -195,6 +213,7 @@ protected void readLastCheckId(){
             String lastIdText = null;
             lastId = -1;
             if ((lastIdText = reader.readLine()) != null) {
+                s_logger.info(String.format("Read [%s] as lastId for Usage sanity checking.", lastIdText));
```
Comment (JoaoJandre) [inline usage/src/main/java/com/cloud/usage/UsageSanityChecker.java:216]:
```suggestion
                logger.info("Read {} as lastId for Usage sanity checking.", lastIdText);
```

File: usage/src/main/java/com/cloud/usage/UsageSanityChecker.java [hunk 2]
```diff
@@ -213,7 +232,9 @@ protected void readMaxId() throws SQLException {
             maxId = -1;
             if (rs.next() && (rs.getInt(1) > 0)) {
                 maxId = rs.getInt(1);
+                s_logger.info(String.format("Read [%s] as maxId for Usage sanity checking.", maxId));
```
Comment (JoaoJandre) [inline usage/src/main/java/com/cloud/usage/UsageSanityChecker.java:235]:
```suggestion
                logger.info("Read {} as maxId for Usage sanity checking.", maxId);
```

File: usage/src/main/java/com/cloud/usage/UsageSanityChecker.java [hunk 3]
```diff
@@ -213,7 +232,9 @@ protected void readMaxId() throws SQLException {
...
                 if (maxId > lastId) {
+                    s_logger.info(String.format("The max id [%s] is greater than the last id [%s]; adding id check to the query.", maxId, lastId));
```
Comment (JoaoJandre) [inline usage/src/main/java/com/cloud/usage/UsageSanityChecker.java:237]:
```suggestion
                    logger.info("The max id {} is greater than the last id {}; adding id check to the query.", maxId, lastId);
```

**Comment by winterhazel** [comment]
@blueorangutan package

**Comment by winterhazel** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9 keepEnv

**Comment by GutoVeronezi** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
I guess it kind of worked because 
```
14:17:18.487 [main] ERROR com.cloud.usage.UsageSanityChecker - Error: Found 919 usage records with raw_usage > 24
```
after running
```
/usr/bin/java -Xmx2G -cp /usr/share/cloudstack-usage/*:/usr/share/cloudstack-usage/lib/*:/usr/share/cloudstack-mysql-ha/lib/*:/etc/cloudstack/usage:/usr/share/java/mysql-connector-java.jar:/usr/share/cloudstack-common com.cloud.usage.UsageSanityChecker
```

unfortunately I had to look up the issue in the code:
```
        addCheckCase("SELECT count(*) FROM `cloud_usage`.`cloud_usage` cu where usage_type not in (4,5) and raw_usage > "
                + aggregationHours,
                "usage records with raw_usage > " + aggregationHours,
                lastCheckId);
```

Out of scope for this PR but I think we can improve there, @winterhazel , cc @vishesh92 .
@JoaoJandre are your concerns met?

**Review by JoaoJandre — APPROVED** [review]
CLGTM, didn't test it

**Pull request merged**